app/entities/user.go

```diff
@@ -47,24 +48,31 @@ func checkName(name string) error {
 
 func checkMail(email string) error {
 	const (
-		maxLocalEmailLen  int = 64
-		maxDomainEmailLen int = 255
+		maxLocalBytes  int = 64
+		maxDomainBytes int = 255
+		regEx              = `(?i)^(?:[a-z\d!#$%&'*+/=?^_\x60{|}~-]+(?:\.[a-z\d!#$%&'*+/=?^_\x60{|}~-]+)*)@(?:(?:[a-z\d](?:[a-z\d-]*[a-z\d])?\.)+[a-z\d](?:[a-z\d-]*[a-z\d])?)$`
 	)
-	// Checking the total length (allowed no more than 64+1+255=320 symbols)
-	emailLen := len(email)
-	if emailLen > maxLocalEmailLen+1+maxDomainEmailLen {
-		return errors.New("email contents too many symbols: " + strconv.Itoa(emailLen))
+	// Checking the lengths of local and domain parts
+	atIndex := strings.IndexByte(email, '@')
+	if atIndex > maxLocalBytes {
+		return fmt.Errorf("local part of email contains too many bytes: %v", atIndex)
 	}
-	// Checking for some email issues by regular expression
-	if valid, _ := regexp.MatchString("(?i)"+`^([^\.@(),:;<>@[\\\]]?[\da-z!#$%&'*+,\-/=?^_\x60{|}~]+.?)+@([^.-][\da-z-]*[^.-]\.?)+([a-z]{2,6})$`, email); !valid {
-		return errors.New("email does not match with regexp \"(?i)\"+`^([^\\.@(),:;<>@[\\\\\\]]?[\\da-z!#$%&'*+,\\-/=?^_\\x60{|}~]+.?)+@([^.-][\\da-z-]*[^.-]\\.?)+([a-z]{2,6})$`")
+	if localPartLen := len(email) - atIndex - 1; localPartLen > maxDomainBytes {
+		return fmt.Errorf("domain part of email contains too many bytes: %v", localPartLen)
+	}
+	// Checking for other email issues by regular expression
+	valid, err := regexp.MatchString(regEx, email)
```

Comment:
regexp.MatchString makes call to compile() each time it's called.
This means, that compile() function will check if you regexp string is correct each time.
To avid that, it's better to make a call to regexp.MustComplile and store result as a Global variable.
So, it will check your regexp string only once.
After that you should do calls to not regexp package but to this variable and call it's method MatchString().

After all. it will look the same. However, instead of package lvl MatchString function you will use MatchString method of an object that will be returned by regexp.MustComplile.

Moreover, please search for an example of it.